scapy/arch/pcapdnet.py

```diff
@@ -372,38 +360,58 @@ def send(self, x):
     conf.L2socket = L2pcapSocket
     conf.L3socket = L3pcapSocket
 
+################
+#  PCAP/PCAPY  #
+################
+
+
 if conf.use_pcap:
     try:
-        import pcap
+        import pcap  # python-pypcap
+        _PCAP_MODE = "pypcap"
     except ImportError as e:
         try:
-            import pcapy as pcap
+            import libpcap as pcap  # python-libpcap
+            _PCAP_MODE = "libpcap"
         except ImportError as e2:
-            if conf.interactive:
-                log_loading.error("Unable to import pcap module: %s/%s", e, e2)
-                conf.use_pcap = False
-            else:
-                raise
+            try:
+                import pcapy as pcap  # python-pcapy
+                _PCAP_MODE = "pcapy"
+            except ImportError as e3:
+                if conf.interactive:
+                    log_loading.error("Unable to import pcap module: %s/%s", e, e2)
+                    conf.use_pcap = False
+                else:
+                    raise
     if conf.use_pcap:
 
         # From BSD net/bpf.h
         # BIOCIMMEDIATE=0x80044270
         BIOCIMMEDIATE = -2147204496
 
-        if hasattr(pcap, "pcap"):  # python-pypcap
+        if _PCAP_MODE == "pypcap":  # python-pypcap
             class _PcapWrapper_pypcap:
-                def __init__(self, device, snaplen, promisc, to_ms):
+                def __init__(self, device, snaplen, promisc, to_ms, monitor=False):
                     try:
-                        self.pcap = pcap.pcap(device, snaplen, promisc, immediate=1, timeout_ms=to_ms)
+                        self.pcap = pcap.pcap(device, snaplen, promisc, immediate=1, timeout_ms=to_ms, rfmon=monitor)
                     except TypeError:
-                        # Older pypcap versions do not support the timeout_ms argument
-                        self.pcap = pcap.pcap(device, snaplen, promisc, immediate=1)
+                        try:
+                            if monitor:
+                                warning("Your pypcap version is too old to support monitor mode, Please update !")
```

Comment:
Can you specify the required version?

Reply:
The fact is, the version supporting that is not merged yet.
https://github.com/pynetwork/pypcap/pull/67/files
They are awaiting our confirmation that it works to merge it, but I can’t test it as I don’t have access anymore to a machine supporting monitor mode

Reply:
Understood.